src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax;
+
+public class Main {
+    public static void main(String[] args) {
+        Figure[] figures1 = new Figure[3];
+        for (int i = 0; i < figures1.length / 2; i++) {
+            Figure figures = new FigureSupplier().getRandomFigure();

```

Comment:
There is no need to create a FigureSuplpier object on each iteration of the loop. Take out in a variable before a cycle and create once. Then use this variable in the loop.